### ISSUE TITLE
litellm missing on streamlit cloud site

### ISSUE THREAD

**Issue by balgad** [haseeb-heaven/LangChain-Coder#17]
I tried to enter OpenAI API key via the streamlit cloud site ([https://langchain-coder.streamlit.app/](https://langchain-coder.streamlit.app/)) but got an error message after pushing enter: "Error loading Open AI: Could not import litellm python package. Please install it with `pip install"

![image](https://github.com/haseeb-heaven/langchain-coder/assets/4057346/ad67a957-c940-4f4a-b8c9-f8dc903cf92d)

Could you fix this?

Thank you

**Comment by haseeb-heaven** [comment]
Okay thanks i will take a look at this issue.

**Comment by haseeb-heaven** [comment]
Okay i have fixed this issue please check

**Comment by balgad** [comment]
It's working now, many thanks!

**Comment by balgad** [comment]
Found another issue:
- after setting up OpenAI with API key
- chose online compiler and accepted privacy policy
- chose different models for chatgpt
- gave an input to coder ("task='write python code using plotly express to draw histograms from a dataframe which has two categorical columns ("a" and "b") and one numerical ("c"). need as many histograms as the number of categorical columns.'") - pressed control+enter after the input
- got an error message after had pushed "execute" button: "Compilation language must not be null"

![image](https://github.com/haseeb-heaven/langchain-coder/assets/4057346/40a2019c-66a3-46b3-91bc-115c3346757a)

**Comment by haseeb-heaven** [comment]
Can you send me the logs of this file and Download them from Settings.

**Comment by balgad** [comment]
Of course, I attach it (now I faced with another issue, seems to have some connection problem, too)
[langchain-coder (2).log](https://github.com/haseeb-heaven/langchain-coder/files/14495083/langchain-coder.2.log)

(anyway, it's a great app! :) )

**Comment by haseeb-heaven** [comment]
> Of course, I attach it (now I faced with another issue, seems to have some connection problem, too) [langchain-coder (2).log](https://github.com/haseeb-heaven/langchain-coder/files/14495083/langchain-coder.2.log)
> 
> (anyway, it's a great app! :) )

Thanks for trying i checked the logs, it seems like you used GPT-3 model, can you try with other GPT Models and see if same error exist.

And yes its great project but **Google gemini Advanced** has **Code Generator** and Editor **IDE** as well so thats what most people will use not.

**Comment by balgad** [comment]
Thanks for your answers.

I checked but something is not okay with online compiler, I attach the log file. After entering all inputs and pushing Generate button an error message appears on seems to be nothing happens because no output at all with online compiler. If I switch to offline compiler the generated code shows up correctly in a lot of cases, but changing back to online seems nothing happens (according to log file I didn't accept policy - but I did it).

Anyway, in EU a lot of employer disable generative AI sites (this happens in my case, too), so quite hard to use them. With your app we got a very good opportunity to use them (and app simplifies generating too in comparison with general prompting).

[langchain-coder (16).log](https://github.com/haseeb-heaven/langchain-coder/files/14497325/langchain-coder.16.log)

**Comment by haseeb-heaven** [comment]
<img width="709" alt="image" src="https://github.com/haseeb-heaven/langchain-coder/assets/11544739/ffeacbdc-b873-4ffa-85bd-0b9d21a4f9d0">

I generated the diagram using my other project [Code-Interpreter](https://github.com/haseeb-heaven/code-interpreter),

Here is the code:

<img width="1652" alt="image" src="https://github.com/haseeb-heaven/langchain-coder/assets/11544739/4bb79c3a-2d53-40f8-802e-55c263eaa351">

The coding Langchain uses online compilers which are limited they dont have all the **external libraries** installed so you need them and its very expensive for me to have servers with lots of space for external libraries so thats only the downside of this product.

I checked the online compiler acting weirdly not setting the language i will try to fix this asap.

**Comment by balgad** [comment]
Thank you for your explanation and help.

Code-interpreter is very promising :) I will try it, it's very likable.

**Comment by haseeb-heaven** [comment]
> Thank you for your explanation and help.
> 
> Code-interpreter is very promising :) I will try it, it's very likable.

Yes Code Interpreter is Python interpreter so any task for Python related which needs external libraries are best suited like Data-analysis and Graphs/Charts generation locally using all LLMs **OpenAI,Gemini,Claude** and more.
